x/ecocredit/client/testsuite/grpc.go

```diff
@@ -543,90 +402,50 @@ func (s *IntegrationTestSuite) TestGetSellOrdersByAddress() {
 			resp, err := rest.GetRequest(tc.url)
-			require.NoError(err)
-
-			var sellOrders ecocredit.QuerySellOrdersByAddressResponse
-			err = val.ClientCtx.Codec.UnmarshalJSON(resp, &sellOrders)
-
-			if tc.expErr {
-				require.Error(err)
-				require.Contains(string(resp), tc.errMsg)
+			require.NoError(err, err)
+			var res marketplace.QuerySellOrdersByBatchDenomResponse
+			err = val.ClientCtx.Codec.UnmarshalJSON(resp, &res)
+			require.NoError(err, err)
+			require.NotNil(res.SellOrders)
+			if tc.expItems > 0 {
+				require.Len(res.SellOrders, tc.expItems)
 			} else {
-				require.NoError(err)
-				require.NotNil(sellOrders.SellOrders)
-				require.Len(sellOrders.SellOrders, tc.expItems)
+				require.Greater(len(res.SellOrders), 0)
 			}
-		})
-	}
-}
-
-func (s *IntegrationTestSuite) TestGetBuyOrder() {
-	val := s.network.Validators[0]
-
-	testCases := []struct {
-		name   string
-		url    string
-		expErr bool
-		errMsg string
-	}{
-		{
-			"not found",
-			fmt.Sprintf("%s/regen/ecocredit/v1alpha1/buy-orders/id/%s", val.APIAddress, "99"),
-			true,
-			"not found",
-		},
-		// TODO: filtered buy orders required #623
-		//{
-		//	"valid request",
-		//	fmt.Sprintf("%s/regen/ecocredit/v1alpha1/buy-orders/id/%s", val.APIAddress, "1"),
-		//	false,
-		//	"",
-		//},
-	}
-
-	require := s.Require()
-	for _, tc := range testCases {
-		tc := tc
-		s.Run(tc.name, func() {
-			resp, err := rest.GetRequest(tc.url)
-			require.NoError(err)
 
-			var buyOrder ecocredit.QueryBuyOrderResponse
-			err = val.ClientCtx.Codec.UnmarshalJSON(resp, &buyOrder)
-
-			if tc.expErr {
-				require.Error(err)
-				require.Contains(string(resp), tc.errMsg)
-			} else {
-				require.NoError(err)
-				require.NotNil(buyOrder.BuyOrder)
-			}
 		})
 	}
 }
 
-func (s *IntegrationTestSuite) TestGetBuyOrders() {
+func (s *IntegrationTestSuite) TestGetSellOrdersByAddress() {
 	val := s.network.Validators[0]
+	_, _, batchDenom := s.createClassProjectBatch(val.ClientCtx, val.Address.String())
+	validAsk := types.NewInt64Coin(core.DefaultParams().AllowedAskDenoms[0].Denom, 10)
+	expiration, err := types2.ParseDate("expiration", "2090-10-10")
+	s.Require().NoError(err)
+	_, err = s.createSellOrder(val.ClientCtx, &marketplace.MsgSell{
+		Owner: val.Address.String(),
+		Orders: []*marketplace.MsgSell_Order{
+			{BatchDenom: batchDenom, Quantity: "10", AskPrice: &validAsk, Expiration: &expiration},
+			{BatchDenom: batchDenom, Quantity: "10", AskPrice: &validAsk, Expiration: &expiration},
+			{BatchDenom: batchDenom, Quantity: "10", AskPrice: &validAsk, Expiration: &expiration},
+		},
+	})
+	s.Require().NoError(err)
 
 	testCases := []struct {
 		name     string
 		url      string
-		expErr   bool
-		errMsg   string
 		expItems int
 	}{
 		{
 			"valid request",
-			fmt.Sprintf("%s/regen/ecocredit/v1alpha1/buy-orders", val.APIAddress),
-			false,
-			"",
-			3,
+			fmt.Sprintf("%s%ssell-orders/address/%s", val.APIAddress, marketplaceRoute, val.Address.String()),
+			-1,
```

Comment:
We should not be expecting a negative number of items.

Reply:
see above